postgres/datadog_checks/postgres/postgres.py

```diff
@@ -560,11 +565,9 @@ def _collect_relations_autodiscovery(self, instance_tags, relations_scopes):
                     for scope in relations_scopes:
                         self._query_scope(cursor, scope, instance_tags, False, db)
         elapsed_ms = (time() - start_time) * 1000
-        self.histogram(
-            "dd.postgres._collect_relations_autodiscovery.time",
-            elapsed_ms,
-            tags=self.tags + self._get_debug_tags(),
-            hostname=self.resolved_hostname,
+        self._report_check_query_perf_metrics(
+            start_time,
+            name='_collect_relations_autodiscovery',
```

Comment:
does it make more sense to have this be a tracked method instead of reporting histogram? example here https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/statement_samples.py#L265

Reply:
it seems like we are inconsistent with this across the check, so im fine with whatever

Reply:
I started with `tracked_method` but had some issue to make it fit into how we query metrics, mainly around adding metric query name. The closest approach is refactor every metric into its own method.